src/preferInlineDecoratorRule.ts

```
@@ -1,79 +1,299 @@
+
+      * \`${OPTION_GETTERS}\` - requires that ${OPTION_GETTERS} are on the same line as its decorator(s). Defaults to \`true\`.
+      * \`${OPTION_METHODS}\` - requires that ${OPTION_METHODS} are on the same line as its decorator(s). Defaults to \`true\`.
+      * \`${OPTION_PARAMETER_PROPERTIES}\` - requires that parameter properties are on the same line as its decorator(s). Defaults to \`true\`.
+      * \`${OPTION_PARAMETERS}\` - requires that ${OPTION_PARAMETERS} are on the same line as its decorator(s). Defaults to \`true\`.
+      * \`${OPTION_PROPERTIES}\` - requires that ${OPTION_PROPERTIES} are on the same line as its decorator(s). Defaults to \`true\`.
+      * \`${OPTION_SETTERS}\` - requires that ${OPTION_SETTERS} are on the same line as its decorator(s). Defaults to \`true\`.
+    `,
+    rationale: 'Placing the decorator on the same line usually makes for shorter code and still easily identifies the declarations.',
     ruleName: 'prefer-inline-decorator',
     type: 'style',
     typescriptOnly: true
   };
 
-  static readonly FAILURE_STRING = 'Consider placing decorators on the same line as the property/method it decorates';
+  static readonly FAILURE_STRING = `Place declarations on the same line as its decorator(s) (${STYLE_GUIDE_LINK})`;
 
   apply(sourceFile: SourceFile): RuleFailure[] {
-    return this.applyWithWalker(new PreferInlineDecoratorWalker(sourceFile, this.getOptions()));
+    const options: OptionDictionary = {
+      ...DEFAULT_OPTIONS,
+      ...this.ruleArguments[0]
+    };
+
+    return this.applyWithFunction(sourceFile, walk, options);
   }
 
   isEnabled(): boolean {
-    return super.isEnabled() && this.ruleArguments.every(ruleArgument => !!(typeof ruleArgument === 'string' && ruleArgument.trim()));
+    return super.isEnabled() && this.areOptionsValid();
+  }
+
+  private areOptionsValid(): boolean {
```

If the options are not valid, we should probably throw an error. Wondering why we're unable to validate directly with the schema? Doesn't tslint do this already?

---

So... this is something that I was about to propose to you: use a lib (something like https://github.com/epoberezkin/ajv) to validate JSON schemas.

AFAIK, there's no in-built validation for tslint. All the checks are "hard-coded" in the `isEnabled` method from https://github.com/palantir/tslint/blob/4b8bbf66b7fe7f49122debdeb02164e045b3a691/src/language/rule/abstractRule.ts#L49

---

What's your opinion about it 👆 ? @wKoza @mgechev

---

JSON schema validation would be great. We do this in the CLI for builders' arguments.